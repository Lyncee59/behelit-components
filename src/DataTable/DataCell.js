import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ${(props) => props.width};
  padding: 1rem;
  box-sizing: border-box;
`

const DataCell = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

DataCell.propTypes = {
  children: PropTypes.node.isRequired,
  width: PropTypes.string.isRequired
}

DataCell.defaultProps = {
  width: 'auto'
}

export default DataCell
