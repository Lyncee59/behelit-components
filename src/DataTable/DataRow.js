import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { theme } from '../Tools/interpolation'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background: ${theme('dataRowBackgroundColor')};
  cursor: ${(props) => props.disableHighlight ? 'default' : 'pointer'};

  &:hover { 
    background: ${(props) => props.disableHighlight ? theme('dataRowBackgroundColor') : theme('dataRowHighlightBackgroundColor')};
  }
`

const DataRow = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    {children}
  </Wrapper>
)

DataRow.propTypes = {
  children: PropTypes.node.isRequired,
  disableHighlight: PropTypes.bool
}

DataRow.defaultProps = {
  disableHighlight: false
}

export default DataRow
