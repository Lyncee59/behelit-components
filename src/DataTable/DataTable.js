import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { theme } from '../Tools/interpolation'

const Wrapper = styled.div`
  width: 100%;
  overflow-x: auto;
`

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  min-width: 1200px;
  padding: 1rem;
  box-sizing: border-box;
  background: ${theme('dataTableBackgroundColor')};
  border: 1px solid ${theme('dataTableBorderColor')};
  border-radius: 5px;

  & > :not(:first-child) {
    border-top: 1px dashed ${theme('dataTableBorderColor')}
  }
`

const DataTable = ({ children, ...rest }) => (
  <Wrapper {...rest}>
    <Container>
      {children}
    </Container>
  </Wrapper>
)

DataTable.propTypes = {
  children: PropTypes.node.isRequired
}

export default DataTable
