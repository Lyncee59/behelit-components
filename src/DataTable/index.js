export { default as DataCell } from './DataCell'
export { default as DataRow } from './DataRow'
export { default as DataTable } from './DataTable'
