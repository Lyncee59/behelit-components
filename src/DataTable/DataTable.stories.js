import React from 'react'
import { storiesOf } from '@storybook/react'

import DataCell from './DataCell'
import DataRow from './DataRow'
import DataTable from './DataTable'

storiesOf('DataTable', module)
  .add('default', () => (
    <DataTable>
      <DataRow><DataCell>1-1</DataCell><DataCell>1-2</DataCell><DataCell>1-3</DataCell></DataRow>
      <DataRow><DataCell>2-1</DataCell><DataCell>2-2</DataCell><DataCell>2-3</DataCell></DataRow>
      <DataRow><DataCell>3-1</DataCell><DataCell>3-2</DataCell><DataCell>3-3</DataCell></DataRow>
    </DataTable>
  ))
  .add('with disableHighlight', () => (
    <DataTable>
      <DataRow disableHighlight><DataCell>1-1</DataCell><DataCell>1-2</DataCell><DataCell>1-3</DataCell></DataRow>
      <DataRow disableHighlight><DataCell>2-1</DataCell><DataCell>2-2</DataCell><DataCell>2-3</DataCell></DataRow>
      <DataRow disableHighlight><DataCell>3-1</DataCell><DataCell>3-2</DataCell><DataCell>3-3</DataCell></DataRow>
    </DataTable>
  ))
  .add('with width', () => (
    <DataTable>
      <DataRow><DataCell width="33%">1-1</DataCell><DataCell width="33%">1-2</DataCell><DataCell width="33%">1-3</DataCell></DataRow>
      <DataRow><DataCell width="33%">2-1</DataCell><DataCell width="33%">2-2</DataCell><DataCell width="33%">2-3</DataCell></DataRow>
      <DataRow><DataCell width="33%">3-1</DataCell><DataCell width="33%">3-2</DataCell><DataCell width="33%">3-3</DataCell></DataRow>
    </DataTable>
  ))
