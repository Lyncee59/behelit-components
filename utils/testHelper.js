import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)