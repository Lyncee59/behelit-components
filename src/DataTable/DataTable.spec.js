import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import DataTable from './DataTable'

describe('DataTable', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<DataTable>MyDataTable</DataTable>)

    expect(tree.find(DataTable)).toMatchSnapshot()
  })
})
