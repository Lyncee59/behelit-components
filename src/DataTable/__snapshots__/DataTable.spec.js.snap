// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render correctly 1`] = `
.c0 {
  width: 100%;
  overflow-x: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-width: 1200px;
  padding: 1rem;
  box-sizing: border-box;
  background: #F3F3F3;
  border: 1px solid #C2C2C2;
  border-radius: 5px;
}

.c1 > :not(:first-child) {
  border-top: 1px dashed #C2C2C2;
}

<DataTable>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 100%;
  overflow-x: auto;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  min-width: 1200px;
  padding: 1rem;
  box-sizing: border-box;
  background: ",
                    [Function],
                    ";
  border: 1px solid ",
                    [Function],
                    ";
  border-radius: 5px;

  & > :not(:first-child) {
    border-top: 1px dashed ",
                    [Function],
                    "
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              MyDataTable
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</DataTable>
`;
